kubespawner/spawner.py

```diff
@@ -1794,11 +1880,12 @@ def _start(self):
             pod = yield gen.maybe_future(self.modify_pod_hook(self, pod))
         for i in range(retry_times):
             try:
-                yield self.asynchronize(
+                created_pod = yield self.asynchronize(
```

Comment:
As I understand it, this created_pod variable will be the return value of the request, equivalent to `kubectl create`, and that it will contain the pod manifest as now known by the k8s api server, which will need to have a `metadata.uid` field.

So therefore, if I'm not understanding this wrong, it doesn't make sense to have the following section below:

```
            yield exponential_backoff(
                lambda: self.is_pod_creating(self.pod_reflector.pods.get(self.pod_name, None)),
                'pod/%s does not exist!' % (self.pod_name),
            )
```

(This is a continuation of https://github.com/jupyterhub/kubespawner/pull/386/files#r404452602)

Reply:
My experience was that the object returned without the UID.